### PR TITLE
Add scrollbar support to simulator sidebar

### LINKED ISSUE #1056: Feature : Add scrollbar to sidebar in simulator
### Is your feature request related to a problem? Please describe

The sidebar in the simulator becomes difficult to navigate when there are many components or options, as it currently does not provide a scrollbar. This affects usability, especially on smaller screens.

### Describe the solution you'd like

Add a vertical scrollbar to the sidebar so users can scroll through all available components easily. This can be implemented using CSS (e.g., overflow-y: auto) to ensure smooth navigation.

### Describe alternatives you've considered

An alternative could be restructuring the sidebar layout to reduce its height or grouping components differently, but adding a scrollbar is a simpler and more effective solution.

### Additional context

This enhancement would improve user experience and make navigation within the simulator more intuitive.

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by aakashgautam-git**
Fixes CircuitVerse/cv-frontend-vue#1056

#### Describe the changes you have made in this PR -

* Added vertical scrollbar support to the simulator sidebar for improved navigation when sidebar content exceeds the available viewport height.
* Improved usability on smaller screen sizes and reduced viewport heights.
* Preserved the existing simulator layout and behavior while introducing scrolling support.
* Avoided nested/double scrollbar behavior and maintained responsive layout handling.

Additionally, the simulator already collapses previously opened component categories when another category is expanded, which helps reduce sidebar height during normal usage. However, on smaller screens the visible sidebar area can still become insufficient, making some components difficult to access. This update ensures the sidebar remains fully navigable in such cases.

Screenshot 2026-05-12 at 12 41 14 AM

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**

* [ ] No, I wrote all the code myself
* [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**

* [x] I have reviewed every single line of the AI-generated code
* [x] I can explain the purpose and logic of each function/component I added
* [x] I have tested edge cases and understand how the code handles them
* [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

The issue was caused by the simulator sidebar becoming difficult to navigate on smaller screens when the available vertical space was insufficient.

To solve this, I added controlled vertical scrolling support using CSS overflow handling while preserving the existing sidebar structure and simulator behavior.

The implementation was intentionally kept minimal to avoid unnecessary UI modifications and to maintain compatibility with the current responsive layout.

Edge cases such as smaller viewport heights, nested scrollbars, and simulator layout preservation were considered during testing.

Water melons are great during summer.

---

## Checklist before requesting a review

* [x] I have added proper PR title and linked to the issue
* [x] I have performed a self-review of my code
* [x] **I can explain the purpose of every function, class, and logic block I added**
* [x] I understand why my changes work and have tested them thoroughly
* [x] I have considered potential edge cases and how my code handles them
* [x] If it is a core feature, I have added thorough tests
* [x] My code follows the project's style guidelines and conventions

## Summary by CodeRabbit

* **Style**
  * Redesigned panel scrolling to use responsive viewport-relative sizing instead of fixed height constraints, enabling better space utilization and improved scrolling experience across different devices.
  * Implemented custom scrollbar styling with unified and consistent visual appearance throughout the panel, including interactive hover state effects that provide enhanced visual feedback during user interactions.

[Review Change Stack](https://app.coderabbit.ai/change-stack/CircuitVerse/cv-frontend-vue/pull/1050)

### CODE REVIEW

File: v1/src/styles/css/main.stylesheet.css [hunk 1]
```diff
@@ -392,6 +392,21 @@ input[type='text']:focus {
...
+    margin-right: 3px;
+    width: 6px;
+}
+.draggable-panel-css .panel-body::-webkit-scrollbar-thumb {
+    background: var(--text-panel);
+    border-radius: 10px;
+}
+.draggable-panel-css .panel-body::-webkit-scrollbar-thumb:hover {
+    background: var(--primary);
 }
```
[the remark this answers is not in the document]
Comment (aakashgautam-git) [reply v1/src/styles/css/main.stylesheet.css:395]: Removed the changes from the legacy `v1/` directory and kept the scrollbar updates only in the active `src/` path.
